flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java

```diff
@@ -545,5 +579,66 @@ public IcebergSource<T> build() {
           table,
           emitter);
     }
+
+    /**
+     * Build the {@link IcebergSource} and create a {@link DataStream} from the source.
+     *
+     * @return data stream from the Iceberg source
+     */
+    public DataStream<T> buildStream(StreamExecutionEnvironment env) {
```

Comment:
This is a new public API. I also thought about the method name as `createStream`. but decided this name for now. open to other suggestion.

also think it is better to require `StreamExecutionEnvironment` here instead of having it a builder method so that it is clear it is not required for the `build()` method.

Reply:
The sentence about `env` is also true for `outputTypeInfo`, and `watermarkStrategy`. Maybe adding them as a parameter would be reasonable too

Reply:
currently, `outputTypeInfo` can be inferred from the `ReaderFunction` if using provided RowData or Avro reader.
```
      if (outputTypeInfo == null) {
        this.outputTypeInfo = inferOutputTypeInfo(table, context, readerFunction);
      }
```

`watermarkStrategy` is defaulted `WatermarkStrategy.noWatermarks`. so it is not mandatory either.

Reply:
`outputTypeInfo` is not needed anymore with the `Converter` interface. Removed watermark strategy for now. We can always add it back in the future if it is needed.